rwengine/src/script/ScriptFunctions.hpp

```diff
@@ -0,0 +1,211 @@
+#ifndef RWENGINE_SCRIPTFUNCTIONS_HPP
+#define RWENGINE_SCRIPTFUNCTIONS_HPP
+
+#include <engine/GameWorld.hpp>
+#include <engine/GameState.hpp>
+#include <engine/GameData.hpp>
+#include <objects/GameObject.hpp>
+#include <script/ScriptTypes.hpp>
+#include <script/ScriptMachine.hpp>
+
+/**
+ * Implementaitions for common functions likely to be shared
+ * among many script modules and opcodes.
+ */
+namespace script {
+const ScriptVec3 kSpawnOffset = ScriptVec3(0.f, 0.f, 1.f);
+
+inline void getObjectPosition(GameObject* object,
+                              ScriptFloat& x,
+                              ScriptFloat& y,
+                              ScriptFloat& z)
+{
+	const auto& p = object->getPosition();
+	x = p.x;
+	y = p.y;
+	z = p.z;
+}
+inline void setObjectPosition(GameObject* object, const ScriptVec3& coord)
+{
+	object->setPosition(coord + script::kSpawnOffset);
+}
+
+inline VehicleObject* getCharacterVehicle(CharacterObject* character)
+{
+	return character->getCurrentVehicle();
```

Comment:
I'm sure this happened before the game could pick a random ped. So it can probably be reproduced by teleporting to the airfield, then toggling the taxi mission.
This happened very often. Similarily the paramedic mission crashed too (when aborting it). I've documented the behaviour on the playable mission tracker as "Crashes in `opcode_0173` on exit because character is nullptr."

I'm not sure if we should fix this in the opcode handler or the common "getVehicle" function and each setter etc. (I'd assume opcode handler because functions should never expect a `nullptr` for `this` or objects they operate on).

It's probably beyond the scope of this PR anyway.